### ISSUE TITLE
Publish  internal/js_*

### ISSUE THREAD

**Issue by 89z** [evanw/esbuild#1635]
would you be willing to move these out of internal, so that they can be imported by other packages? I am looking for a JavaScript parser and/or lexer for general use.

**Comment by evanw** [comment]
No, I'm not willing to do that. The APIs for the parser and lexer are specific to esbuild and not intended for general use. You will need to either fork and maintain your own copy of the code or find another project instead.

**Comment by 89z** [comment]
Bummer :(

I found some other options, but they all seem to have their own problems. Will keep looking.

**Comment by 89z** [comment]
> You will need to either fork and maintain your own copy of the code

actually I might do this. Does the package have some Walk type function, like this?

https://godocs.io/github.com/tdewolff/parse/v2/js#Walk

**Comment by evanw** [comment]
No. It isn't really needed because esbuild only ever walks the AST twice. It's pretty trivial to create such a function, however.
